nixos/lib/test-driver/lib/driver.py

```diff
@@ -0,0 +1,175 @@
+from contextlib import contextmanager
+from typing import Iterator, Optional
+import atexit
+import os
+import ptpython.repl
+import pty
+import subprocess
+import sys
+import tempfile
+import traceback
+
+import common
+
+from machine import Machine, NixStartScript
+
+
+class VLan ():
+    def __init__(self, nr: int, tmp_dir: str):
+        self.nr = nr
+        self.socket_dir = os.path.join(tmp_dir, f"vde{self.nr}.ctl")
```

Comment:
might as well use the more modern pathlib.Path here (and anywhere else relevant)

Reply:
done :+1: also `.format()` -> `f""`